src/models/hooks/objective.js

```diff
@@ -201,12 +202,16 @@ const propogateStatusToParentGoal = async (sequelize, instance, options) => {
 
       // and if so, we update it (storing the previous status so we can revert if needed)
       if (atLeastOneInProgress) {
-        await goal.update({
-          status: 'In Progress',
-          previousStatus: 'Not Started',
-        }, {
+        // eslint-disable-next-line global-require
+        const changeGoalStatus = require('../../goalServices/changeGoalStatus').default;
+        const userId = httpContext.get('impersonationUserId') || httpContext.get('loggedUser');
```

Comment:
question, could the parameter order of changeGoalStatus be changed, so that userId is near the end, then have the httpContext calls used as the default value if no user is supplied. That way it would not need to be included all over the pace and need to be called/looked up all over, when its only needed in the one place for the changeGoalStatus function?

Reply:
`changeGoalStatus` only has one parameter: an object